pil-analyzer/src/expression_processor.rs

```diff
@@ -174,20 +174,17 @@ impl<D: AnalysisDriver> ExpressionProcessor<D> {
 
     pub fn process_function(
         &mut self,
-        params: &[String],
+        params: &[Pattern],
         expression: ::powdr_ast::parsed::Expression,
     ) -> Expression {
         let previous_local_vars = self.save_local_variables();
 
-        // Add the new local variables, potentially overwriting existing variables.
-        self.local_variables.extend(
-            params
-                .iter()
-                .zip(self.local_variable_counter..)
-                .map(|(p, i)| (p.clone(), i)),
-        );
-        self.local_variable_counter += params.len() as u64;
-
+        for param in params {
+            if !param.is_irrefutable() {
+                panic!("Function parameters must be irrefutable, but {param} is refutable.");
```

Comment:
we should really clean up these panics...

Reply:
You mean use Results instead? I think as long as we don't have source references in sub-expressions, we cannot display a nice error message anyway, so it's fine to use panic  for now

Reply:
yea I know, just meant one day

Reply:
![image](https://github.com/powdr-labs/powdr/assets/504195/e881b029-62b1-480d-be1f-5559851711d0)